### PR TITLE
[BUGFIX] Multidomain typoscript configuration

### LINKED ISSUE #841: Multi domain breaks hooks from ItemsProcFunc.php
## Description

If you use more than one root page (multi domain) in your installation, it is possible that the lists "show these facets" and "fields for extended search" inside the configuration of the search plugin are empty. These lists are build with hooks from ItemsProcFunc.php, which needs the typoscript configuration to extract the storage pid and build up these lists. But in case of a multi domain installation, the typoscript configuration is restricted to a page tree and the default storage pid is used, which could be a different page tree than the plugin is placed.

## Reproduction

Steps to reproduce the behaviour:

1. Go to the typo3 backend
2. Place a new page on root
3. Mark this page as root page
4. Go to the search plugin
5. Open plugin tab

## Expected Behavior

It must be possible to configure the page tree where the typoscript configuration should be read from, or it must depend on the plugin placement. Otherwise, it is not possible to define different configurations for each page tree. In our case, we have three different configurations. Two of them are on the same page tree and the other one is on a different tree.

### PR DESCRIPTION
**Pull request by christopher-timm**
Read the typoscript configuration depending on the page where it is placed on.
Otherwise, multi domain installation only get the default site root typoscript.
This PR is linked to issue #841

### CODE REVIEW

File: Classes/Hooks/ItemsProcFunc.php [hunk 1]
```diff
@@ -50,19 +50,38 @@ public function toolList(&$params)
...
+            $ts = $objectManager->get(\TYPO3\CMS\Core\TypoScript\TemplateService::class,[$siterootRow['uid']]);
+            $ts->rootLine = $rootline;
+            $ts->runThroughTemplates($rootline, 0);
+            $ts->generateConfig();
+        } catch (\Exception $e) {
+            die($e->getMessage());
+        }
+
+        $typoscriptConfig = $ts->setup;
+        $this->storagePid = $typoscriptConfig["plugin."]["tx_dlf."]["persistence."]["storagePid"];
```
Comment (sebastian-meyer) [inline Classes/Hooks/ItemsProcFunc.php]:
```suggestion
        $this->storagePid = $typoscriptConfig['plugin.']['tx_dlf.']['persistence.']['storagePid'];
```

File: Classes/Hooks/ItemsProcFunc.php [hunk 2]
```diff
@@ -50,19 +50,38 @@ public function toolList(&$params)
...
+            }
+        }
+
+        try {
+            $ts = $objectManager->get(\TYPO3\CMS\Core\TypoScript\TemplateService::class,[$siterootRow['uid']]);
+            $ts->rootLine = $rootline;
+            $ts->runThroughTemplates($rootline, 0);
+            $ts->generateConfig();
+        } catch (\Exception $e) {
+            die($e->getMessage());
```
Comment (sebastian-meyer) [inline Classes/Hooks/ItemsProcFunc.php]: Please use proper error handling here instead of just exiting the script.

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]

**Comment by christopher-timm** [comment]
Requested changes should be fixed now

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]
Please check the codacy issues. Apart from that, everything looks good, now!

**Review by sebastian-meyer — DISMISSED** [review]
Perfect! I can merge this as soon as you bring your branch up to date with the base branch.

(When opening a pull request you can choose to allow me to commit changes to your branch. If you do so, I can update your branch myself.)

**Review by sebastian-meyer — APPROVED** [review]

**Pull request merged**